google/cloud/pubsub/internal/tracing_message_batch_test.cc

```diff
@@ -56,6 +57,17 @@ void EndSpans(std::vector<opentelemetry::nostd::shared_ptr<
   }
 }
 
+/// Creates @p n spans.
+std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
+CreateSpans(int n) {
+  std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
+      spans(n);
+  std::generate(spans.begin(), spans.end(), [i = 0]() mutable {
+    return MakeSpan("test span " + std::to_string(i++));
+  });
+  return spans;
+}
+
```

Comment:
`auto` return should work in this case:

https://godbolt.org/z/aj7641Y1d

```suggestion
auto CreateSpans(int n) {
  std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
      spans(n);
  std::generate(spans.begin(), spans.end(), [i = 0]() mutable {
    return MakeSpan("test span " + std::to_string(i++));
  });
  return spans;
}
```